### DISCUSSION TITLE
how about internal player volume controller by scrolling on systray icon?

### DISCUSSION THREAD

**Discussion by mhandze** [strawberrymusicplayer/strawberry#862] [Ideas]
Hello,

Mamy of media players can change their internal software volumen when you scroll mouse on their systray icon. This volume is independent from main volume level and it's quick acess when you want to turn level of music down for a moment.

If that makes sense then let's (I can) create new request on pull.

thanks!

**Comment by jonaski** [comment]
The reason I removed it was because it wasn't working reliable on all desktop environments and users were filing issues for it, some periodically not working and some not working at all. It's fine with me if you agree to test that it works on all major Linux Desktops (KDE, Gnome, XFCE, LXQT, Cinnamon, MATE) + Windows and macOS.

**Comment by mhandze** [comment]
I have access to KDE and Windows 10. I can test here without problem.

Why dont make it option unchecked (by default) and even called experimental in brackets.

That way it could bring more feedback from users.

**Comment by jonaski** [comment]
OK. I can test on XFCE if you mange to get it to work.
Looks like I removed it in 2021 as part of this commit: https://github.com/strawberrymusicplayer/strawberry/commit/67f831bebacd396f1b9f3bbdef8ef94de461570d#diff-2ac6ecef0e06ee0b4e8a25402956b1a7fa56e3fec31319cc8fba513422ed24a8
From what I remember it was broken on all Linux desktops I tested on, maybe still working on Windows.

I tested adding the code back (no need to use eventfilter anymore), but I can't get any event to trigger on either XFCE or KDE, tested with both the latest development branch of Qt 6 and Qt 5.15. (I think it still works on Windows though, but I didn't test now).

Code:

```
diff --git a/src/core/qtsystemtrayicon.cpp b/src/core/qtsystemtrayicon.cpp
index ba136383..8ed7b3fb 100644
--- a/src/core/qtsystemtrayicon.cpp
+++ b/src/core/qtsystemtrayicon.cpp
@@ -204,3 +204,50 @@ void SystemTrayIcon::LoveVisibilityChanged(const bool value) {
 void SystemTrayIcon::LoveStateChanged(const bool value) {
   if (action_love_) action_love_->setEnabled(value);
 }
+
+bool SystemTrayIcon::event(QEvent *event) {
+
+  qLog(Debug) << __PRETTY_FUNCTION__;
+
+  if (QObject::event(event)) return true;
+
+  if (event->type() == QEvent::Wheel) {
+    QWheelEvent *e = static_cast<QWheelEvent*>(event);
+    if (e->modifiers() == Qt::ShiftModifier) {
+      if (e->angleDelta().y() > 0) {
+        emit SeekForward();
+      }
+      else {
+        emit SeekBackward();
+      }
+    }
+    else if (e->modifiers() == Qt::ControlModifier) {
+      if (e->angleDelta().y() < 0) {
+        emit NextTrack();
+      }
+      else {
+        emit PreviousTrack();
+      }
+    }
+    else {
+      QSettings s;
+      s.beginGroup(BehaviourSettingsPage::kSettingsGroup);
+      bool prev_next_track = s.value("scrolltrayicon").toBool();
+      s.endGroup();
+      if (prev_next_track) {
+        if (e->angleDelta().y() < 0) {
+          emit NextTrack();
+        }
+        else {
+          emit PreviousTrack();
+        }
+      }
+      else {
+        emit ChangeVolume(e->angleDelta().y());
+      }
+    }
+  }
+
+  return true;
+
+}
diff --git a/src/core/qtsystemtrayicon.h b/src/core/qtsystemtrayicon.h
index 7021882a..6252d657 100644
--- a/src/core/qtsystemtrayicon.h
+++ b/src/core/qtsystemtrayicon.h
@@ -64,6 +64,9 @@ class SystemTrayIcon : public QSystemTrayIcon {
   bool MuteEnabled() const { return action_mute_->isVisible(); }
   void SetMuteEnabled(const bool enabled) { action_mute_->setVisible(enabled); }
 
+  protected:
+   bool event(QEvent *event) override;
+
  private:
   QPixmap CreateIcon(const QPixmap &icon, const QPixmap &grey_icon);
   void UpdateIcon();

```

**Comment by jonaski** [comment]
Looks like there is a bug report for it here:
https://bugreports.qt.io/browse/QTBUG-91697
